internal/controller/linstorcluster_controller.go

```diff
@@ -1010,13 +1030,87 @@ func (r *LinstorClusterReconciler) reconcileClusterState(ctx context.Context, lc
 		err = lc.Controller.Modify(ctx, *modification)
 		if err != nil {
 			conds.AddError(conditions.Configured, err)
-			return err
+			return status, err
 		}
 	}
 
 	conds.AddSuccess(conditions.Configured, "Properties applied")
 
-	return r.reconcileCSINodes(ctx, lcluster, lc, conds)
+	err = r.getClusterStatus(ctx, lcluster, lc, &status)
+	if err != nil {
+		conds.AddError(conditions.Configured, err)
+		return status, err
+	}
+
+	return status, r.reconcileCSINodes(ctx, lcluster, lc, conds)
+}
+
+func (r *LinstorClusterReconciler) getClusterStatus(ctx context.Context, lcluster *piraeusiov1.LinstorCluster, lc *linstorhelper.Client, status *piraeusiov1.LinstorClusterStatus) error {
+	var satelliteList piraeusiov1.LinstorSatelliteList
+	err := r.Client.List(ctx, &satelliteList, client.MatchingLabels{
+		"app.kubernetes.io/instance": lcluster.Name,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to fetch LinstorSatellites: %w", err)
+	}
+
+	var scheduled, running int32
+	for _, satellite := range satelliteList.Items {
+		if satellite.DeletionTimestamp == nil {
+			scheduled++
+		}
+
+		cond := meta.FindStatusCondition(satellite.Status.Conditions, string(conditions.Available))
+		if cond != nil && cond.Status == metav1.ConditionTrue {
+			running++
+		}
+	}
+
+	status.ScheduledSatellites = &scheduled
+	status.RunningSatellites = &running
+
+	sp, err := lc.Nodes.GetStoragePoolView(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to fetch storage pools: %w", err)
+	}
+
+	var total, free int64
+	for _, pool := range sp {
+		if pool.TotalCapacity == math.MaxInt64 {
+			// Skip all the "diskless" pools, they always report max capacity
+			continue
+		}
+
+		// LINSTOR reports KiB
+		total += pool.TotalCapacity * 1024
+		free += pool.FreeCapacity * 1024
```

Comment:
It would be neat if we could use linstor's dedicated available storage API, rather than adding up all the storage pools. But that applies to resource groups, so we would have to add it to the storage class. I guess that is not possible.